### PR TITLE
Tutorial Bugfixes

### PR DESCRIPTION
**Pull request by yolanother**
When importing samples it brings in the asmdefs as well. We want samples to import cleanly so removing the asmdefs.

### CODE REVIEW

**Comment by yolanother** [comment]
Added a few fixes for bugs discovered while working on tutorials.

File: Scripts/Runtime/WitRequest.cs [hunk 1]
```diff
@@ -47,6 +47,23 @@ public enum Endian
...
+        /// <summary>
+        /// Error code thrown when an exception is caught during processing or
+        /// some other general error happens that is not an error from the server
+        /// </summary>
+        public const int ERROR_CODE_GENERAL = -1;
+
+        /// <summary>
+        /// Error code returned when no configuration is defined
+        /// </summary>
+        public const int ERROR_CODE_NO_CONFIGURATIOON = -2;
```
Comment (nick-benson) [inline Scripts/Runtime/WitRequest.cs:59]: Typo

File: Scripts/Editor/Utility/WitUnderstandingViewer.cs [hunk 2]
```diff
@@ -53,6 +54,26 @@ static void Init()
...
+                if (wit)
+                {
+                    wit.events.OnResponse.AddListener((r) =>
+                    {
+                        response = r;
+                        var u = r["text"].Value;
+                        if (!string.IsNullOrEmpty(u))
+                        {
+                            utterance = u;
+                        }
```
Comment (nick-benson) [inline Scripts/Editor/Utility/WitUnderstandingViewer.cs:71]: Could get rid of single-variable names here by doing

```
if (!string.IsNullOrEmpty(response["text"].Value))
{
    utterance = response["text"].Value;
}
```

although that duplicates the unwrapping.

File: Scripts/Runtime/CallbackHandlers/MultiValueHandler.cs [hunk 3]
```diff
@@ -73,15 +73,26 @@ protected override void OnHandleResponse(WitResponseNode response)
                         formatEvent.onFormattedValueEvent?.Invoke(result);
                     }
                 }
+            }
 
-                List<string> values = new List<string>();
-                for (int i = 0; i < references.Length; i++)
+            // We want to invoke if at least one value was found. Anything that
+            // was missing will be empty, but we must have at least one value
+            // in the response for the invoke to happen.
```
Comment (nick-benson) [inline Scripts/Runtime/CallbackHandlers/MultiValueHandler.cs:80]: Does Wit consider something to be a match if one part of it matches and the others are empty? Should we have "required" and "non-required" parameters? This change here seems like a nontrivial change of meaning.

**Review by nick-benson — APPROVED** [review]

File: Scripts/Editor/Utility/WitUnderstandingViewer.cs (hunk 2 above)
Comment (yolanother) [reply Scripts/Editor/Utility/WitUnderstandingViewer.cs:71]: Yeah, that duplicates the lookup for the value for text. Maybe I should add a TryGetValue to the node.

File: Scripts/Runtime/CallbackHandlers/MultiValueHandler.cs (hunk 3 above)
Comment (yolanother) [reply Scripts/Runtime/CallbackHandlers/MultiValueHandler.cs:80]: This was the intended meaning of this callback for this utility component. As long as one entity returns a value you get the callback. The array structure is always the same. The true match is on the intent, so maybe I should have this callback fire no matter what if the intent is matched and let the dev decide what to do with no entities.

**Pull request merged**